### DISCUSSION TITLE
libcurl crash inside Curl_llist_insert_next

### DISCUSSION THREAD

**Discussion by shivakumargunda** [curl/curl#6468] [Q&A] [answered]
Curl 7.74.0 crashes inside Curl_llist_insert_next. This crash seen consistently after every (6 to 9) hrs of appliation run. 
I suspect libcurl crashes while trying to handle burst(~72) of http requests. This issue reproducible on centos Linux with openSSL and c-ares packages installed.

Previously I was using curl 7.63.0, this same crash was not observed at all but a different crash inside Curl_llist_remove() observed. Both these crashes points to the linked list.  Something went wrong in managing linked list.

Pasted below backtrace of both the crashes(7.74.0 & 7.63.0).

I am trying to create a example which reproduces this issue. 

Please help me to understand what went wrong.

Backtrace(Version: 7.74.0):
#0  0x00007fb7a68821ea in Curl_llist_insert_next ()
#1  0x00007fb7a6876d4c in Curl_hash_add () 
#2  0x00007fb7a688996e in sh_addentry () 
#3  0x00007fb7a688d8c0 in singlesocket () 
#4  0x00007fb7a688e0a0 in multi_socket () 
#5  0x00007fb7a688e7ae in curl_multi_socket_action ()

Backtrace(Version: 7.63.0):
#0 0x00007f0d01018cd1 in ?? ()
#1 0x00007f0d3bf28a49 in Curl_llist_remove () 
#2 0x00007f0d3bf2906f in Curl_hash_delete () 
#3 0x00007f0d3bf2980f in sh_delentry () 
#4 0x00007f0d3bf2d6b3 in Curl_multi_closed () 
#5 0x00007f0d3bf283f3 in Curl_closesocket () 
#6 0x00007f0d3bf143ae in conn_free () 
#7 0x00007f0d3bf14ac2 in Curl_disconnect () 
#8 0x00007f0d3bf2a085 in multi_done () 
#9 0x00007f0d3bf2cb2a in multi_runsingle () 
#10 0x00007f0d3bf2da06 in multi_socket () 
#11 0x00007f0d3bf2e35e in curl_multi_socket_action ()

**Comment by bagder** [comment, marked as answer]
Continue at https://github.com/curl/curl/issues/6469
